cli/lib/metadata.ts

```diff
@@ -31,11 +32,13 @@ export async function injectMetadata(jobUri: string) {
     throw new Error(`Dataset ${project.dataset} not loaded`)
   }
 
+  const datasetTriples = dataset.pointer.dataset.match(null, null, null, dataset.id)
+
   return obj(function (quad: Quad, _, callback) {
-    const visited = new Set()
+    const visited = new TermSet()
     const copyChildren = (subject: QuadObject) => {
       if (subject && subject.termType !== 'Literal' && !visited.has(subject)) {
-        [...dataset.pointer.dataset.match(subject)].forEach(item => {
+        [...datasetTriples.match(subject, null, null, dataset.id)].forEach(item => {
```

Comment:
Not necessary since you filter out the correct triples above

```suggestion
        [...datasetTriples.match(subject)].forEach(item => {
```